docs/index.html

```diff
@@ -20,396 +20,490 @@
                 } catch (error) {
-                    console.error("Error creating annotations folder:", error);
+                    console.error("Error updating file list:", error);
+                } finally {
+                    setLoading(false);
                 }
-                // Run code to start annotation session
+            };
+
+            const mountFolder = async () => {
                 try {
-                    // await runCode();
-                    // TODO: get annotator_url from stdout
-                    // Placeholder for setting annotationURL
-                    setAnnotationURL("http://example.com/annotation-session-url"); // Replace with actual URL
+                    const dirHandle = await window.showDirectoryPicker();
+                    setImageFolderHandle(dirHandle);
                 } catch (error) {
-                    console.error("Error running Python code:", error);
+                    console.error("Error accessing folder:", error);
                 }
-            }
-        };
+            };
+
+            useEffect(() => {
+                if (imageFolderHandle) {
+                    window.saveToDB('imageFolderDirHandle', imageFolderHandle);
+                    updateFileList(imageFolderHandle, setImageList, setIsLoadingImages); // Update image list
+                    console.log("Mounted folder:", imageFolderHandle.name);
+                }
+            }, [imageFolderHandle]);
+
+            // const checkAnnotationURL = async () => {
+            //     const intervalId = setInterval(() => {
+            //         const url = localStorage.getItem('annotationURL');
+            //         if (url) {
+            //             setAnnotationURL(url);
+            //             clearInterval(intervalId); // Stop checking once the URL is available
+            //         }
+            //     }, 1000);
+            // };
 
-        const shareAnnotationURL = async () => {
-            if (!annotationURL) return;
+            const deployAnnotationSession = async () => {
+                if (imageFolderHandle) {
+                    try {
+                        const annotationsFolder = await imageFolderHandle.getDirectoryHandle('annotations', { create: true });
+                        setAnnotationsFolderHandle(annotationsFolder);
+                        await updateFileList(annotationsFolder, setAnnotationsList, setIsLoadingAnnotations);
+                        console.log("Created annotations folder:", annotationsFolder.name);
+                    } catch (error) {
+                        console.error("Error creating annotations folder:", error);
+                    }
+                    // Run code to start annotation session
+                    const code = await fetch(`./${REGISTER_SERVICE_FILE}`).then(response => response.text());
+                    try {
+                        setIsRunning(true);
+                        await workerPromise;
```

Comment:
This is when we actually wait for the worker ready (it start loading when the page loaded, and keep loading in the background).